pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1004,6 +998,51 @@ func (in ResourcesDuration) IsZero() bool {
 	return len(in) == 0
 }
 
+type WorkflowConditions []*WorkflowCondition
+
+func (wc *WorkflowConditions) AddOrReplaceCondition(condition *WorkflowCondition) {
+	for index, wfCondition := range *wc {
+		if wfCondition.Type == condition.Type {
+			(*wc)[index] = condition
+			return
+		}
+	}
+	*wc = append(*wc, condition)
+}
+
+func (wc *WorkflowConditions) AddOrAppendCondition(condition *WorkflowCondition) {
```

Comment:
I found this func confusing and had to inspect the source code to understand - this is about appending the message yes? Maybe a name that reflects that?